src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx

```diff
@@ -77,101 +101,380 @@ export const Confirm: React.FC<ConfirmProps> = ({ accountId, onNext }) => {
   )
 
+  const selectedCurrency = useAppSelector(selectSelectedCurrency)
+
+  // notify
+  const toast = useToast()
+
+  useEffect(() => {
+    ;(async () => {
+      if (!(accountId && opportunityData?.stakedAmountCryptoBaseUnit && asset)) return
+
+      const amountCryptoBaseUnit = bnOrZero(state?.withdraw.cryptoAmount).times(
+        bn(10).pow(asset.precision),
+      )
+
+      if (amountCryptoBaseUnit.isZero()) return
+
+      const amountCryptoThorBaseUnit = toThorBaseUnit({
+        valueCryptoBaseUnit: amountCryptoBaseUnit,
+        asset,
+      })
+
+      const withdrawBps = getWithdrawBps({
+        withdrawAmountCryptoBaseUnit: amountCryptoBaseUnit,
+        stakedAmountCryptoBaseUnit: opportunityData.stakedAmountCryptoBaseUnit,
+        rewardsamountCryptoBaseUnit: opportunityData?.rewardsAmountsCryptoBaseUnit?.[0] ?? '0',
+      })
+
+      const quote = await getThorchainSaversWithdrawQuote({ asset, accountId, bps: withdrawBps })
+
+      const { dust_amount, expected_amount_out } = quote
+      const withdrawFee = amountCryptoThorBaseUnit
+        .minus(expected_amount_out)
+        .div(`1e+${THOR_PRECISION}`)
+
+      const dustAmountCryptoPrecision = fromThorBaseUnit(dust_amount)
+
+      setWithdrawFee(withdrawFee.toFixed())
+      setDustAmount(dustAmountCryptoPrecision.toFixed(asset.precision))
+    })()
+  }, [
+    accountId,
+    asset,
+    opportunityData?.rewardsAmountsCryptoBaseUnit,
+    opportunityData?.stakedAmountCryptoBaseUnit,
+    state?.withdraw.cryptoAmount,
+  ])
+
+  const getMaybeUtxoAccountAddress: () => Promise<string> = useCallback(async () => {
+    if (!accountId) throw new Error('accountId is undefined')
+    if (!isUtxoChainId(chainId)) return ''
+
+    try {
+      const position = await getThorchainSaversPosition({ accountId, assetId })
+      debugger
+      const { asset_address } = position
+      const accountAddress = chainId === bchChainId ? `bitcoincash:${asset_address}` : asset_address
+
+      return accountAddress
+    } catch (_e) {
+      throw new Error(`Cannot get savers position for accountId: ${accountId}`)
+    }
+  }, [accountId, assetId, chainId])
+
+  const getEstimateFeesArgs: () => Promise<EstimateFeesInput> = useCallback(async () => {
+    if (!(accountId && opportunityData?.stakedAmountCryptoBaseUnit?.[0]))
+      throw new Error('accountId is undefined')
+
+    const amountCryptoBaseUnit = bnOrZero(state?.withdraw.cryptoAmount).times(
+      bn(10).pow(asset.precision),
+    )
+
+    const withdrawBps = getWithdrawBps({
+      withdrawAmountCryptoBaseUnit: amountCryptoBaseUnit,
+      stakedAmountCryptoBaseUnit: opportunityData?.stakedAmountCryptoBaseUnit,
+      rewardsamountCryptoBaseUnit: opportunityData?.rewardsAmountsCryptoBaseUnit?.[0] ?? '0',
+    })
+    const quote = await getThorchainSaversWithdrawQuote({ asset, accountId, bps: withdrawBps })
+
+    const maybeUtxoAccountAddress = await getMaybeUtxoAccountAddress()
```

Comment:
prob need to guard against this not existing at runtime

Reply:
d172d88a6cd573fcd24836e27b2ec2a35501ff19